internal/sdkv2provider/resource_cloudflare_list_item.go

```diff
@@ -0,0 +1,217 @@
+	client := meta.(*cloudflare.API)
+	accountID := d.Get(consts.AccountIDSchemaKey).(string)
+	listID := d.Get("list_id").(string)
+
+	listItem, err := client.GetListItem(ctx, cloudflare.AccountIdentifier(accountID), listID, d.Id())
+	if err != nil {
+		if strings.Contains(err.Error(), "could not find item") {
+			tflog.Info(ctx, fmt.Sprintf("List item %s no longer exists", d.Id()))
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(errors.Wrap(err, fmt.Sprintf("error reading List Item with ID %q", d.Id())))
+	}
+
+	d.Set("comment", listItem.Comment)
+
+	if listItem.IP != nil {
+		d.Set("ip", *listItem.IP)
+	}
+
+	if listItem.Redirect != nil {
+		optBoolToString := func(b *bool) string {
+			if b != nil {
+				switch *b {
+				case true:
+					return "enabled"
+				case false:
+					return "disabled"
+				}
+			}
+			return ""
+		}
+
+		d.Set("source_url", listItem.Redirect.SourceUrl)
+		d.Set("include_subdomains", optBoolToString(listItem.Redirect.IncludeSubdomains))
+		d.Set("target_url", listItem.Redirect.TargetUrl)
+		d.Set("status_code", listItem.Redirect.StatusCode)
+		d.Set("preserve_query_string", optBoolToString(listItem.Redirect.PreserveQueryString))
+		d.Set("subpath_matching", optBoolToString(listItem.Redirect.SubpathMatching))
+		d.Set("preserve_path_suffix", optBoolToString(listItem.Redirect.PreservePathSuffix))
+	}
+
+	return nil
+}
+
+func resourceCloudflareListItemUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	// The API does not currently support in-place update of list items
+	dr := resourceCloudflareListItemDelete(ctx, d, meta)
+	if dr != nil {
+		return dr
+	}
+
+	cr := resourceCloudflareListItemCreate(ctx, d, meta)
+	if cr != nil {
+		return cr
+	}
+
+	return resourceCloudflareListItemRead(ctx, d, meta)
+}
+
+func resourceCloudflareListItemDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+	accountID := d.Get(consts.AccountIDSchemaKey).(string)
+	listID := d.Get("list_id").(string)
+
+	_, err := client.DeleteListItems(ctx, cloudflare.AccountIdentifier(accountID), cloudflare.ListDeleteItemsParams{
+		ID: listID,
+		Items: cloudflare.ListItemDeleteRequest{
+			Items: []cloudflare.ListItemDeleteItemRequest{{ID: d.Id()}},
+		},
+	})
+	if err != nil {
+		return diag.FromErr(errors.Wrap(err, fmt.Sprintf("error removing List Item %s from list %s", listID, d.Id())))
+	}
+
+	return nil
+}
+
+func listItemType(d *schema.ResourceData) string {
+	if _, ok := d.GetOk("ip"); ok {
+		return "ip"
+	}
+
+	return "redirect"
+}
+
+func buildListItemCreateRequest(d *schema.ResourceData) cloudflare.ListItemCreateRequest {
+	itemType := listItemType(d)
+
+	request := cloudflare.ListItemCreateRequest{
+		Comment: d.Get("comment").(string),
+	}
+
+	if itemType == "ip" {
+		request.IP = cloudflare.StringPtr(d.Get("ip").(string))
+		return request
+	}
+
+	// itemType must be 'redirect'
```

Comment:
let's remove this comment as it doesn't seem to be adding any value here. generally, we follow the guidelines at https://stackoverflow.blog/2021/12/23/best-practices-for-writing-code-comments/ for when to keep comments in code.